sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -163,6 +163,15 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    // If the database name is not provided, fallback to v1 catalog.
+    case ShowTablesStatement(None, pattern) =>
+      ShowTablesCommand(None, pattern)
+
+    case ShowTablesStatement(Some(CatalogObjectIdentifier(Some(catalog), ident)), pattern) =>
+      ShowTables(catalog.asTableCatalog, ident, pattern)
+
+    case ShowTablesStatement(Some(db), pattern) =>
+      ShowTablesCommand(Some(db.mkString(".")), pattern)
```

Comment:
This should be used if no default v2 catalog is set.

Reply:
This is updated. Thanks.